packages/amplify-cli-core/src/types.ts

```diff
@@ -201,7 +391,15 @@ export type GetPackageAssetPaths = () => Promise<string[]>;
 /**
  * Placeholder type
  */
-export type $IPluginManifest = $TSAny;
+export type $IPluginManifest = {
+  name: string;
+  type: string;
+  commands?: string[];
+  services?: string[];
+  functionRuntime?: {
+    pluginId: string;
+  };
```

Comment:
Does every plugin define `functionRuntime` ? What does `functionRuntime` mean in this context?
If this is plugin specific we should find a way to subtype.

Reply:
It's not clear what this references. There is a `FunctionRuntime` defined here: https://github.com/sdstolworthy/amplify-cli/blob/f4d6faa880fd4b316672a04e54a7c13aa9440a37/packages/amplify-function-plugin-interface/src/index.ts#L180. However, there is not a `pluginId` defined on that interface.

I hesitate to assume those are actually intended to be the same type. As far as I can tell, `functionRuntime` in this context is only ever referenced here: https://github.com/sdstolworthy/amplify-cli/blob/f4d6faa880fd4b316672a04e54a7c13aa9440a37/packages/amplify-cli/src/extensions/amplify-helpers/load-runtime-plugin.ts#L7.

Reply:
Meanwhile we bumped into this with @evcodes . It seems to be related to `XXX-function-runtime-provider` packages. See https://github.com/aws-amplify/amplify-cli/pull/11985 for a bit of breadcrumbs.
The function runtime providers seem to form their own plugin family (plugins within plugins?...).
Now I wonder if `$IPluginManifest` is only for categories or "any plugin"... within cli. I.e. category function is a plugin that has runtime plugins inside...
I'd say that lumping any plugin into single type isn't cohesive and we should probably have two types `CategoryPlugin` and `FunctionRuntimeProvider` (or so).

Reply:
https://github.com/sdstolworthy/amplify-cli/blob/tscontext-typing/packages/amplify-cli/src/extensions/amplify-helpers/read-breadcrumbs.ts#L20

This file shows the shape that structure is supposed to have. Thanks for the tip. I've created a type in `types.ts` to describe the shape. I think that distinction will work for the purposes of this PR.